inbm/integration-reloaded/scripts/create_signature.py

```diff
@@ -18,12 +18,12 @@
     if num_params == 4:
         password = sys.argv[3].encode('utf-8')
 
-with open(package_name) as package:
-    checksum = hashlib.sha384(package.read().encode('utf-8')).hexdigest()
+with open(package_name, 'rb') as package:
+    checksum = hashlib.sha384(package.read()).hexdigest()
```

Comment:
**Suggestion:** Consider reading the file content in chunks to handle large files which might not fit into memory if read all at once. [performance]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
with open(package_name, 'rb') as package:
    sha384_hash = hashlib.sha384()
    for chunk in iter(lambda: package.read(4096), b""):
        sha384_hash.update(chunk)
    checksum = sha384_hash.hexdigest()
```